lib/index.ml

```diff
@@ -71,18 +87,38 @@ let get_job_ids t ~owner ~name ~hash =
   | Sqlite3.Data.[ TEXT variant; TEXT id ] -> variant, Some id
   | row -> Fmt.failwith "get_job_ids: invalid row %a" Db.dump_row row
 
-let record ~repo ~hash jobs =
+module Status_cache = struct
+  let cache = Hashtbl.create 1_000
+  let cache_max_size = 1_000_000
+
+  type elt = [ `Not_started | `Pending | `Failed | `Passed ]
+
+  let add ~owner ~name ~hash (status : elt) =
+    if Hashtbl.length cache > cache_max_size then Hashtbl.clear cache;
```

Comment:
Perhaps it would be better to crash here... This status information will not (necessarily) be re-generated if cache entries are invalidated, but the service will catch up eventually.

Reply:
If we index by ref rather than by hash then it won't grow so much anyway, and we could remove old entries in `set_active_refs` if necessary. That would probably mean returning the status in `RefInfo` rather than from the commit, though.